### ISSUE TITLE
[Android Bootcamp] Project 0: Todo - Ready for Review 

### ISSUE THREAD

**Issue by sharathp** [sharathp/SimpleToDo#2]
SimpleToDo implementation is complete, please review. 

/cc @codepathreview @codepath

**Comment by sharathp** [comment]
SimpleToDo implementation is complete, please review, thanks!! (In case you missed my previous comment)

/cc @codepathreview @codepath

**Comment by nesquena** [comment]
Sharath,

Looks good, this exercise is intended in part to give you an introduction to the general rhythm of this course. The course is entirely project-based with an app being assigned each week and then due the following week. Each project builds on the last to help each engineer  learn the practical elements of Android development and best practices as quickly as possible. We also do a code review for each submitted project once the program begins.

The next step is to continue working on [extensions to your todo app](http://courses.codepath.com/snippets/intro_to_android/prework#heading-5-extending-your-todo) and to **schedule a short 10 minute phone conversation** with us. Navigate to [today's date and onward from there](https://calendar.google.com/calendar/selfsched?sstoken=UURvT01vbzBza1pnfGRlZmF1bHR8NzVjYzBkN2EyZTRiOGRmYjZlZjhmMjA2N2IyMGMwYWU) and choose an available 15-minute slot in the mornings or evenings. Let us know if none of those times work.

With regards to extensions to implement, we are looking for developers with an interest in user experience and product development so be sure to focus on additional functional enhancements and/or improving the user interface of the app.

Once you select a slot, can you make sure to **include the best number** to reach you at to the invite? Look forward to chatting soon!

**Comment by sharathp** [comment]
Hello @nesquena, thanks for reviewing.  Sure, will work on extending the app.  Looking forward to talk to you.

**Comment by sharathp** [comment]
Enhanced the initial version, added the following features:
- Suggested: Persist the todo items into SQLite instead of a text file
- Suggested: Improve style of the todo items in the list using a custom adapter
- Suggested: Add support for completion due dates for todo items (and display within listview item)
- Additional: Add support for selecting the priority of each todo item (and display in listview item)
- Additional: Tweak the style improving the UI / UX, play with colors, images or backgrounds 
- Additional: Provided Date & Priority pickers via DialogFragment (Instead of editing items via DialogFragment as a toolbar is not usual for DialogFragments)
- Additional: By default the list doesn't include the past items. Provided mechanism to toggle showing/hiding past items. Also provided a mechanism to delete all items
- Additional: Added description field to todo item

Also updated the demo gif in README.md (https://raw.githubusercontent.com/sharathp/SimpleToDo/master/simpletodo_demo_2.gif)

(Here is the tag for reference - https://github.com/sharathp/SimpleToDo/releases/tag/v2.0)

I want to push a couple of additional enhancements as time permits.  Thanks! Please review!

/cc @codepathreview @codepath @nesquena

**Comment by sharathp** [comment]
please let me know your comments/thoughts.  thanks

@codepathreview @codepath @nesquena

**Comment by sharathp** [comment]
Further enhancements:
- Navigation Drawer (Provides the ability to toggle between Todays Items, All Items & High Priority Items options)
- Prompt before deleting all items
- Recurring notification at 4 AM everyday

Also updated the demo gifs in README.md (https://raw.githubusercontent.com/sharathp/SimpleToDo/master/simpletodo_demo_3.gif & https://raw.githubusercontent.com/sharathp/SimpleToDo/master/simpletodo_demo_3_notif.gif)

(Here is the tag for reference - https://github.com/sharathp/SimpleToDo/releases/tag/v3.0)

Thanks! Please review!

/cc @codepathreview @codepath @nesquena

**Comment by nesquena** [comment]
Great, thanks for the update and glad to see you so many optionals implemented. Your pre-work looks great! We will be determining final admittance into the program over the next couple weeks. For those not admitted as participants, we will offer access to applicants as a remote observer cohort that has access to the materials and is able to work on the projects at their own pace. You will be hearing from us again soon.

**Comment by sharathp** [comment]
Thanks for the update @nesquena!  I want to make more enhancements but I understand that it is already past the due date,  but I would like to know if any further enhancements to the app from now onwards would still be considered for admission evaluation?  Thanks!

**Comment by nesquena** [comment]
@sharathp Yes, the initial deadline is referring to initial pre-work submissions that will be processed. You should feel free to continue working on your todo app if you'd like and all enhancements are considered as we make our final determinations. Thanks for checking.

**Comment by sharathp** [comment]
More enhancements: (v4)
- Added ability to search in all ToDo Items (via "All Items" section in navigation drawer, searches name and description fields)
- Added ability to share a ToDo Item as plain text.
- Added ability to see Overdue Items (via Notification drawer)
- Added ability to delete completed Items

Also updated the demo gifs in README.md 

(Here is the tag for reference - https://github.com/sharathp/SimpleToDo/releases/tag/v4.0)

Thanks! Please review!

/cc @codepathreview @codepath @nesquena

**Comment by nesquena** [comment]
Nice job with this! Your pre-work looks excellent, hope you feel you were able to learn something through this even given all of your existing Android experience.

**Comment by sharathp** [comment]
@nesquena - Yes, definitely! this did help me learn some aspects that I missed previously, I am very excited and looking forward to go through this bootcamp, I am pretty sure I will learn best practices and also plenty of other stuff after I attend this bootcamp and share with the community. I cannot wait enough to hear about your final list of participants! Please let me know if you have any questions/comments regarding the code, design etc.. Thanks!

**Comment by sharathp** [comment]
Hello @nesquena, i see that there is an open slot today night around 10 PM EST, is it ok if I take that slot just to have a quick chat with you.  (want to remind you that we talked once a couple of weeks ago)

**Comment by nesquena** [comment]
Hi Sharath, sure feel free to book the open slot if you'd like to chat. We will be determining admissions soon and we have been keeping track of your pre-work so you will be hearing back from us soon via email.

**Comment by sharathp** [comment]
Sounds good! I went ahead and booked that slot, looking forward to talk to you! Thanks!

**Comment by sharathp** [comment]
Enhancements: (v5)
- FAB appear/disappear on scrolling up and down
- List -> Detail explode transition
- Swipe to delete list items 

Updated demo gif in README.md

(Here is the tag for reference - https://github.com/sharathp/SimpleToDo/releases/tag/v5.0)

Thanks! Please review!

/cc @codepathreview @codepath @nesquena

**Comment by nesquena** [comment]
👍  Nice! Look forward to getting started next week!
